libs/dh/admin/feature-invite-user-modal/src/lib/dh-invite-user-modal.component.ts

```diff
@@ -117,65 +131,81 @@ export class DhInviteUserModalComponent implements AfterViewInit, OnDestroy {
 
   ngAfterViewInit(): void {
     this.inviteUserModal.open();
-    this.actorIdSubscription = this.userInfo.controls.actorId.valueChanges.subscribe((actorId) => {
-      actorId !== null
-        ? this.userInfo.controls.email.enable()
-        : this.userInfo.controls.email.disable();
-
-      if (actorId === null) {
-        this.actorStore.resetOrganizationState();
-        return;
+
+    this.userEmailExistsQuery.valueChanges.pipe(takeUntil(this.destroy$)).subscribe((x) => {
+      this.knownEmails = x.data?.knownEmails?.map((x) => x.toUpperCase()) ?? [];
+      this.isLoadingEmails = false;
+      this.changeDectorRef.detectChanges();
+    });
+
+    this.baseInfo.controls.actorId.valueChanges
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((actorId) => {
+        actorId !== null
+          ? this.baseInfo.controls.email.enable()
+          : this.baseInfo.controls.email.disable();
+
+        if (actorId === null) {
+          this.actorStore.resetOrganizationState();
+          return;
+        }
+
+        this.assignableUserRolesStore.getAssignableUserRoles(actorId);
+        this.actorStore.getActorOrganization(actorId);
+      });
+
+    this.actorStore.organizationDomain$.pipe(takeUntil(this.destroy$)).subscribe((domain) => {
+      this.domain = domain;
+    });
+
+    this.baseInfo.controls.email.valueChanges
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(async (email) => {
```

Comment:
suggestion: unnecessary `async`


```suggestion
      .subscribe((email) => {
```